vespalib/src/tests/net/tls/capabilities/capabilities_test.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/vespalib/net/tls/capability_set.h>
+#include <vespa/vespalib/testkit/test_kit.h>
+
+using namespace vespalib;
+using namespace vespalib::net::tls;
+using namespace std::string_view_literals;
+
+TEST("Capability bit positions are stable across calls") {
+    auto cap1 = Capability::content_storage_api();
+    auto cap2 = Capability::content_storage_api();
+    EXPECT_EQUAL(cap1.id_bit_pos(), cap2.id_bit_pos());
+}
+
+TEST("Capability instances are equality comparable") {
+    auto cap1 = Capability::content_document_api();
+    auto cap2 = Capability::content_document_api();
+    auto cap3 = Capability::content_storage_api();
+    EXPECT_EQUAL(cap1, cap2);
+    EXPECT_EQUAL(cap2, cap1);
+    EXPECT_NOT_EQUAL(cap1, cap3);
+}
+
+TEST("Can get underlying name of all Capability instances") {
+    EXPECT_EQUAL(Capability::content_storage_api().name(),  "vespa.content.storage_api"sv);
+    EXPECT_EQUAL(Capability::content_document_api().name(), "vespa.content.document_api"sv);
+    EXPECT_EQUAL(Capability::content_search_api().name(),   "vespa.content.search_api"sv);
+    EXPECT_EQUAL(Capability::slobrok_api().name(),          "vespa.slobrok.api"sv);
+    EXPECT_EQUAL(Capability::content_status_pages().name(), "vespa.content.status_pages"sv);
+    EXPECT_EQUAL(Capability::content_metrics_api().name(),  "vespa.content.metrics_api"sv);
+    EXPECT_EQUAL(Capability::content_cluster_controller_internal_state_api().name(),
+                 "vespa.content.cluster_controller.internal_state_api"sv);
+}
+
+TEST("Capability instances can be stringified") {
+    EXPECT_EQUAL(Capability::content_storage_api().to_string(), "Capability(vespa.content.storage_api)");
+}
+
+TEST("All known capabilities can be looked up by name") {
```

Comment:
consider also matching the values against the actual capabilities

Reply:
Added checking of all returned values